### PR TITLE
Add "Next" target to UI Makefile

### PR DESCRIPTION
**Pull request by mrtracy**
Commit adds a "next" target to the UI makefile, which provides an effective and
simple mechanism for isolating a planned overhaul of the UI.

This is an alternative to working on a separate fork or branch; I believe that
this will help keep progress on this UI version more visible to all contributors.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/5936)

### CODE REVIEW

File: ui/Makefile [hunk 1]
```diff
@@ -128,15 +146,21 @@ $(GOBINDATA_TARGET): $(GOBINDATA_DEPS) bower.installed release/$(INDEX)
 	rm -f $(INDEX)
 	cp release/$(INDEX) $(INDEX)
 	chmod a-w $(INDEX)
-	$(GO_BINDATA) -nometadata -pkg ui -o $@ $(GOBINDATA_DEPS) $(INDEX) $(GOBINDATA_SOURCES)
+	$(GO_BINDATA) -nometadata -pkg ui -o $@ $(GOBINDATA_DEPS) $(INDEX) $(INDEX_NEXT) $(GOBINDATA_SOURCES)
```
Comment (tamird) [inline ui/Makefile]: INDEX_NEXT isn't defined

Comment (mrtracy) [inline ui/Makefile]: Removed

**Comment by tamird** [comment]
Reviewed 4 of 5 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[ui/Makefile, line 46 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5936#-KEmaZjg5dGiBWAuhdr5:-KEmaZjg5dGiBWAuhdr6:-424731710) ([raw file](https://github.com/cockroachdb/cockroach/blob/67a7ec94d2e40c9698a44cee12e319967fd4683d/ui/Makefile#L46)):*
is there any reason not to call this file app.js?

---

*[ui/next/build/app_next.js, line 0 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5936#-KEmaQ73clKbzP2akQ-m:-KEmaQ73clKbzP2akQ-n:-1766435410) ([raw file](https://github.com/cockroachdb/cockroach/blob/5685a92fe68bdd5d7281fd3f31f3bcb5026f575d/ui/next/build/app_next.js#L0)):*
i think you want to gitignore next/build

---

*[ui/next/ts/app.ts, line 2 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5936#-KEmaJ6Eq1mS5DwPX3Yw:-KEmaJ6Eq1mS5DwPX3Yx:1357011281) ([raw file](https://github.com/cockroachdb/cockroach/blob/67a7ec94d2e40c9698a44cee12e319967fd4683d/ui/next/ts/app.ts#L2)):*
i think you can just call this `Admin`. you're already in the next directory, yeah?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/5936)*

**Comment by mrtracy** [comment]
Your .gitignore comment is somehow hidden from me in the reviewable UI; it has been addressed.

---

Review status: 4 of 5 files reviewed at latest revision, 3 unresolved discussions.

---

*[ui/Makefile, line 46 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5936#-KEmaZjg5dGiBWAuhdr5:-KEmi_LqyMxCr5qnZbUX:-1482562876) ([raw file](https://github.com/cockroachdb/cockroach/blob/67a7ec94d2e40c9698a44cee12e319967fd4683d/ui/Makefile#L46)):*
I suppose not, changed.

---

*[ui/next/ts/app.ts, line 2 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5936#-KEmaJ6Eq1mS5DwPX3Yw:-KEmihNYZHhT7wX1ypNm:-1147142271) ([raw file](https://github.com/cockroachdb/cockroach/blob/67a7ec94d2e40c9698a44cee12e319967fd4683d/ui/next/ts/app.ts#L2)):*
Meh, this is a placeholder, it'll be gone on the very next commit.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/5936#-:-KEmk0Da2j4K0h2oXQKV:-36993054)*

**Comment by tamird** [comment]
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[ui/Makefile, line 46 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5936#-KEmaZjg5dGiBWAuhdr5:-KEn8HcH2GlHh2RwFzZs:-573646570) ([raw file](https://github.com/cockroachdb/cockroach/blob/67a7ec94d2e40c9698a44cee12e319967fd4683d/ui/Makefile#L46)):*
doesn't look changed

---

*[ui/next/ts/app.ts, line 2 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5936#-KEmaJ6Eq1mS5DwPX3Yw:-KEn8LNkg2s-bxaadbMB:5174689) ([raw file](https://github.com/cockroachdb/cockroach/blob/67a7ec94d2e40c9698a44cee12e319967fd4683d/ui/next/ts/app.ts#L2)):*
ok

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/5936)*

**Comment by mrtracy** [comment]
Review status: 4 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[ui/Makefile, line 46 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5936#-KEmaZjg5dGiBWAuhdr5:-KEwyyWIrB6NXYgPGm7g:-1638213950) ([raw file](https://github.com/cockroachdb/cockroach/blob/67a7ec94d2e40c9698a44cee12e319967fd4683d/ui/Makefile#L46)):*
Hmm, got dropped somehow.  ACTUALLY fixed.

---

*[ui/next/build/app_next.js, line  \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5936#-KEmaQ73clKbzP2akQ-m:-KEwz2_fpu5nfj8O3B3E:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5685a92fe68bdd5d7281fd3f31f3bcb5026f575d/ui/next/build/app_next.js#L)):*
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/5936)*

**Comment by tamird** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/5936#-:-KF4jBDTDCDSiWBH91Jn:1417088721)*

**Pull request merged**